### PR TITLE
Prepare functionality to show `HomeScreenAlert` for iOS devices

### LINKED ISSUE #604: [Push Notifications] Mobile iOS Set Home Icon screen is missing
**Describe the bug**
For iOS devices only, we want to show this [Add to Home Screen UI](https://www.figma.com/file/Cm6qATC1CpK9nUrKgx5K3U/Notifications?type=design&node-id=94-1480&mode=design&t=3vz4RENcLToKuHMw-0) This is a modal which should appear before the Enable Push Notifications Modal.

### PR DESCRIPTION
**Pull request by shelegdmitriy**
Closes #604

Related to [#367](https://github.com/near/near-discovery-components/pull/367)

[Loom video](https://www.loom.com/share/11fc20ec8e064f49b69a75ef76cda257?sid=4ca31428-3a36-4644-b624-0cf507c486c8)

Do not merge before:

- [x] https://github.com/near/near-discovery-components/pull/367
- [x] https://github.com/near/near-discovery-components/pull/383

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -120,6 +150,14 @@ const HomePage: NextPageWithLayout = () => {
             onOpenChange: handleModalCloseOnEsc,
           }}
         />
+        <VmComponent
+          // src={components.nearOrg.notifications.iosHomeScreenAlert}
+          src="dima_sheleg.near/widget/NearOrg.Notifications.HomeScreenAlert"
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:155]: To test it locally use `bos-loader` and change this `src`

File: src/pages/index.tsx [hunk 2]
```diff
@@ -103,6 +120,19 @@ const HomePage: NextPageWithLayout = () => {
...
+    }
+  }, [iOSDevice]);
+
+  useEffect(() => {
+    if (iOSDevice) {
+      window.matchMedia('(display-mode: standalone)').addEventListener('change', (e) => setHomeScreenApp(e.matches));
+    }
+  }, [iOSDevice]);
+
+  console.log("iOSDevice: ", iOSDevice, "isHomeScreenApp: ", isHomeScreenApp);
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:135]: I'll clean this up later

File: src/pages/index.tsx [hunk 3]
```diff
@@ -103,6 +120,19 @@ const HomePage: NextPageWithLayout = () => {
...
+    if (iOSDevice) {
+      setHomeScreenApp(window.matchMedia('(display-mode: standalone)').matches);
+    }
+  }, [iOSDevice]);
+
+  useEffect(() => {
+    if (iOSDevice) {
+      window.matchMedia('(display-mode: standalone)').addEventListener('change', (e) => setHomeScreenApp(e.matches));
+    }
+  }, [iOSDevice]);
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:133]: These lines checks does user uses a home screen app. Here is a [doc](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/display-mode)

File: src/pages/index.tsx [hunk 4]
```diff
@@ -57,19 +69,24 @@ const HomePage: NextPageWithLayout = () => {
...
-          setShowNotificationModalState(showNotificationModal());
-        }, 10000);
+        // let's test it on preview first and then I can change that
+        if (iOSDevice && !isHomeScreenApp) {
+          setIosHomeScreenPrompt(true);
+        } else {
+          setTimeout(() => {
+            setShowNotificationModalState(showNotificationModal());
+          }, 10000);
+        }
```
Comment (shelegdmitriy) [inline src/pages/index.tsx:79]: As I mention in my loom video we can change this

Comment (charleslavon) [reply src/pages/index.tsx:79]: This is 10 seconds?  What happens if the user clicks to a different screen before 10 seconds has passed?  I'm thinking perhaps this should be 5 seconds.

File: src/pages/index.tsx (hunk 3 above)
Comment (charleslavon) [reply src/pages/index.tsx:133]: Is it necessary that lines 125 and 131 exist in distinct useEffects?

File: src/pages/index.tsx (hunk 1 above)
Comment (charleslavon) [reply src/pages/index.tsx:155]: Let's [merge this](https://github.com/near/near-discovery-components/pull/367) to facilitate testing.

File: src/pages/index.tsx (hunk 4 above)
Comment (charleslavon) [reply src/pages/index.tsx:79]: 3 👍🏾

File: src/pages/index.tsx (hunk 3 above)
Comment (shelegdmitriy) [reply src/pages/index.tsx:135]: Yes. This is the same functionality as for [Navigation](https://github.com/near/near-discovery/blob/develop/src/components/navigation/Navigation.tsx#L9-L15) component where we checking mobile/desktop device

**Review by charleslavon — APPROVED** [review]

**Pull request merged**